src/triage/component/postmodeling/contrast/model_evaluator.py

```diff
@@ -0,0 +1,1199 @@
+                                figsize=(16, 12),
+                                fontsize=20):
+        """
+        Plot recall and precision curves against depth into the list.
+        """
+
+
+        _labels = self.predictions.filter(items=['label_value', 'score'])
+
+        y_score = _labels.score
+        precision_curve, recall_curve, pr_thresholds = \
+            metrics.precision_recall_curve(_labels.label_value, y_score)
+
+        precision_curve = precision_curve[:-1]
+        recall_curve = recall_curve[:-1]
+
+        # a little bit of a hack, but ensure we start with a cut-off of 0
+        # to extend lines to full [0,1] range of the graph
+        pr_thresholds = np.insert(pr_thresholds, 0, 0)
+        precision_curve = np.insert(precision_curve, 0, precision_curve[0])
+        recall_curve = np.insert(recall_curve, 0, recall_curve[0])
+
+        pct_above_per_thresh = []
+        number_scored = len(y_score)
+        for value in pr_thresholds:
+            num_above_thresh = len(y_score[y_score >= value])
+            pct_above_thresh = num_above_thresh / float(number_scored)
+            pct_above_per_thresh.append(pct_above_thresh)
+        pct_above_per_thresh = np.array(pct_above_per_thresh)
+
+        plt.clf()
+        fig, ax1 = plt.subplots(figsize=figsize)
+        ax1.plot(pct_above_per_thresh, precision_curve, "#000099")
+        ax1.set_xlabel('Proportion of population', fontsize=fontsize)
+        ax1.set_ylabel('Precision', color="#000099", fontsize=fontsize)
+        plt.ylim([0.0, 1.05])
+        ax2 = ax1.twinx()
+        ax2.plot(pct_above_per_thresh, recall_curve, "#CC0000")
+        ax2.set_ylabel('Recall', color="#CC0000", fontsize=fontsize)
+        plt.xlim([0.0, 1.0])
+        plt.ylim([0.0, 1.05])
+        plt.title("Precision-recall at x-proportion", fontsize=fontsize)
+        plt.show()
+
+    def _error_labeler(self,
+                      path,
+                      param=None,
+                      param_type=None):
+        '''
+        Explore the underlying causes of errors using decision trees to explain the
+        residuals base on the same feature space used in the model. This
+        exploration will get the most relevant features that determine y - y_hat
+        distance and may help to understand the outomes of some models.
+
+        This function will label the errors and return two elements relevant to
+        model these. First, a feature matrix (X) with all the features used by
+        the model. Second, an iterator with different labeled errors: FPR, FRR,
+        and the general error.
+
+        Arguments:
+            - param_type: (str) type of parameter to define a threshold. Possible
+              values come from triage evaluations: rank_abs, or rank_pct
+            - param: (int) value
+            - path: path for the ProjectStorage class object
+        '''
+        self.preds_matrix(path)
+        test_matrix = self.preds_matrix(path)
```

Comment:
ditto

Reply:
If I call the `self.preds_matrix(path)` method once, the function will retrieve the matrix and store it in the class `dict`, but will not return anything. Once I make that call, I can access the stored data in a second call to the object `self.preds_matrix`. I think this is another bad use of the `@cachedproperty` feature and this should remain as a normal method. 

What do you think @jesteria?

Reply:
Sorry for misusing your crafted tools, btw

Reply:
What? I am looking at your definition of `preds_matrix` right now. It's not a `cachedproperty`, it's a method, in which we implemented a custom dictionary cache. 

There is zero reason it *couldn't* return the matrix the first time, and every time, the same, regardless of whether it was internally computed or retrieved from a cache.

And, in fact, looking at it now, it clearly does return the matrix, regardless.

This code is clearly unnecessary -- you can just remove that first line, wherever you call `preds_matrix`.